lib/modules/datasource/docker/index.ts

```diff
@@ -558,6 +617,80 @@ export class DockerDatasource extends Datasource {
     return null;
   }
 
+  @cache({
+    namespace: 'datasource-docker-architecture',
+    key: (
+      registryHost: string,
+      dockerRepository: string,
+      currentDigest: string
+    ) => `${registryHost}:${dockerRepository}@${currentDigest}`,
+    ttlMinutes: 1440 * 28,
+  })
+  public async getImageArchitecture(
+    registryHost: string,
+    dockerRepository: string,
+    currentDigest: string
+  ): Promise<string | null | undefined> {
+    try {
+      const manifestResponse = await this.getManifestResponse(
+        registryHost,
+        dockerRepository,
+        currentDigest,
+        'head'
+      );
+
+      if (
+        manifestResponse?.headers['content-type'] === MediaType.manifestV2 ||
+        manifestResponse?.headers['content-type'] === MediaType.ociManifestV1
+      ) {
+        const configDigest = await this.getConfigDigest(
+          registryHost,
+          dockerRepository,
+          currentDigest
+        );
+        if (!configDigest) {
+          logger.debug(
+            {
+              registryHost,
+              dockerRepository,
+              currentDigest,
+            },
+            'Unexpected error while retrieving config digest for docker image'
+          );
+          return undefined;
+        }
+
+        const configResponse = await this.getImageConfig(
+          registryHost,
+          dockerRepository,
+          configDigest
+        );
+        if (configResponse) {
+          const imageConfiguration = JSON.parse(configResponse.body);
+          const architecture =
+            (imageConfiguration['architecture'] as string) ?? null;
+          logger.debug(
+            `Current digest ${currentDigest} relates to architecture ${
+              architecture ?? 'null'
+            }`
+          );
+
+          return architecture;
+        }
+      }
+    } catch (err) /* istanbul ignore next */ {
+      if (err instanceof ExternalHostError) {
+        throw err;
+      }
+      logger.debug(
+        { registryHost, dockerRepository, currentDigest, err },
+        'Unknown error getting image architecture'
+      );
+    }
```

Comment:
same here, remove error handling completly

Reply:
Not sure about this one because it would result in a different, so far **unintended** code flow. 

If `this.getImageConfig()` (= `GET /library/some-dep/blobs/some-config-digest`) returns 404 the handling is:
- Currently: `architecture` will be _undefined_ and the manifest digest will be used instead
- With removed error handling: An exception in https://github.com/renovatebot/renovate/blob/026d5c677591e191b0b98d07408b875dc7111003/lib/modules/datasource/docker/index.ts#L958 will immediately trigger the catch clause in `getDigest()` and there's no fallback to the manifest digest.

IMHO it makes no sense to fail completely, if there is a reasonable fallback with the manifest digest that works independent from the image config.

Keep it or kill the fallback?

Reply:
Then only handle 404 and rethrow any other error.